Ripped from Fedora

* Thu Dec 22 2005 Karel Zak <kzak@redhat.com> 3.1.5-5
- fix "gawk -v BINMODE=1" (patch by Aharon Robbins)

        * eval.c (set_BINMODE): Fix logic of test for no numeric value.
        Makes `gawk -v BINMODE=1 ...' work again. Thanks to Eli Zaretskii
        <eliz@gnu.org> for pointing out the problem.

--- gawk-3.1.5/eval.c.binmode	2005-12-22 19:05:32.000000000 +0100
+++ gawk-3.1.5/eval.c	2005-12-22 19:06:40.000000000 +0100
@@ -2167,7 +2167,7 @@
 			}
 		}
 
-		if (! digits || (BINMODE_node->var_value->flags & MAYBE_NUM) == 0) {
+		if (! digits && (BINMODE_node->var_value->flags & MAYBE_NUM) == 0) {
 			BINMODE = 0;
 			if (strcmp(p, "r") == 0)
 				BINMODE = 1;
